Add Nested Route Example to README.md (#359)
preactjs/preact-router: README.md

:information_desk_person: Pages are just regular components that get mounted when you navigate to a certain URL.
Any URL parameters get passed to the component as `props`.

> Defining what component(s) to load for a given URL is easy and declarative.  
> You can even mix-and-match URL parameters and normal `props`.  
> You can also make params optional by adding a `?` to it.  
> When you add `*` or `+` to param, it matches to match the path to the left of param.  
> `*` matches when `partical_path` is 0 or more characters.  
> `+` matches when `partical_path` is 1 or more characters.

Reviewer: ```suggestion
Defining what component(s) to load for a given URL is easy and declarative.
Querystring and `:parameter` values are passed to the matched component as props.
 Parameters can be made optional by adding a `?`, or turned into a wildcard match by adding `*` (zero or more characters) or `+` (one or more characters):
```

```js
<Router>
  <A path="/" />
  <B path="/b" id="42" />
  <C path="/c/:id" />
  <C path="/d/:optional?/:params?" />

>   <D path="/e/:partical_path*" />
>   <E path="/f/:partical_path+" />

Reviewer: ```suggestion
  <D path="/e/:remaining_path*" />
  <E path="/f/:remaining_path+" />
```

route('/page-3', true)  // replaces the current history entry
```

> ### Nested Route Example

Reviewer: ```suggestion
### Nested Routers

The `<Router>` is a self-contained component that renders based on the page URL. When nested a Router inside of another Router, the inner Router does not share or observe the outer's URL or matches. Instead, inner routes must include the full path to be matched against the page's URL:
```

> ```js
> import { h, render } from 'preact'
> import Router from 'preact-router'
> import { Link } from 'preact-router/match'
> 
> function Foo() {
>     return (
>       <div>
>         <h1>Foo</h1>
>         <Router>
>           <Bar path="/foo/bar" />
>           <Baz path="/foo/baz" />
>         </Router>
>       </div>
>     )
> }
> 
> function Bar() {
>   return <h2>Bar</h2>
> }
> 
> function Baz() {
>   return <h2>Baz</h2>
> }
> 
> 
> function App() {
>   return (
>     <div>
>       <Router>
>         <Foo path="/foo/:partical_path*" />
>       </Router>
>       <Link activeClassName="active" href="/foo">Foo</Link><br />
>       <Link activeClassName="active" href="/foo/bar">Bar</Link><br />
>       <Link activeClassName="active" href="/foo/baz">Baz</Link>
>     </div>
>   )
> }
> 
> render(<App />, document.body)
> ```

Reviewer: ```suggestion
```js
import { h, render } from 'preact';
import Router from 'preact-router';

function Profile(props) {
    // `props.rest` is the rest of the URL after "/profile/"
    return (
      <div>
        <h1>Profile</h1>
        <Router>
          <MyProfile path="/profile/me" />
          <UserProfile path="/profile/:user" />
        </Router>
      </div>
    );
}
const MyProfile = () => (<h2>My Profile</h2>);
const UserProfile = (props) => (<h2>{props.user}</h2>);

function App() {
  return (
    <div>
      <Router>
        <Home path="/" />
        <Profile path="/profile/:rest*" />
      </Router>
      <nav>
        <a href="/">Home</a>
        <a href="/profile/me">My Profile</a>
        <a href="/profile/alice">Alice's Profile</a>
      </nav>
    </div>
  );
}

render(<App />, document.body);
```
```
